src/app/model/ChessEngine.ts

```diff
@@ -1,3 +1,48 @@
 import { ChessModel } from './ChessModel';
+import { Side, Square } from './Types';
+import { Piece } from './Piece';
+import { Player } from './Player';
+import { ChessBoard } from './ChessBoard';
+import { PieceWasMoved } from './PieceWasMoved';
+import { PieceWasCaptured } from './PieceWasCaptured';
 
-export class ChessEngine implements ChessModel {}
+export class ChessEngine implements ChessModel {
+  currentSide: Side = Side.BLACK;
+
+  constructor(readonly board: ChessBoard) {}
+
+  move(byPlayer: Player, squareFrom: Square, squareTo: Square): (PieceWasMoved | PieceWasCaptured)[] {
+    const chosenPiece = this.board.onPositionPiece(squareFrom) as Piece;
+    const possibleMoves = chosenPiece.possibleMoves(squareFrom, this.board);
+    const pieceOnSquare = this.board.onPositionPiece(squareTo);
+    const isOponent = pieceOnSquare && pieceOnSquare.side !== chosenPiece.side;
```

Comment:
Ta logikę używamy dojśc często i możn ją nawet dac do Piece. Np. Piece.isOpponentOf(anotherPiece: Piece), co Ty na to :) ?